contents/javascript/Javascript-framework-Angular.md

```diff
@@ -0,0 +1,204 @@
+ 
+2. **Angular uses TypeScript.** Angular applications are built using TypeScript language, a superset of JavaScript. TypeScript provides static typing. This helps the compiler show warnings about any potential errors in the code, before the app runs. Another advantage is that TypeScript provides code completion using [Intellisense](https://docs.microsoft.com/en-us/visualstudio/ide/using-intellisense?view=vs-2017). IntelliSense provides active hints as code is added.
+
+    <box> TypeScript was created and is maintained by Microsoft. It has been <a href="https://developer.telerik.com/featured/the-rise-of-typescript/">growing in popularity</a> over the last few years making it reliable to use.</box>
+                                                  
+3. **Google Long-Term Support.** Some software engineers consider the fact that Angular is supported by Google an advantage of the technology. While this may sound justified, it does not serve as much of an incentive. The good sign though is that Google announced Long-Term Support (LTS) for the technology back in 2017.
+
+    These are some of the reasons why you might want to get familiar with Angular.
+
+    [source](https://hackr.io/blog/why-should-you-learn-angular-in-2018)
+
+### Benefit 1: Angular provides a declarative user interface.
+
+Angular uses HTML to define the app’s user interface. HTML is a declarative language which is more intuitive than using imperative code to define the UI. An Imperative approach requires you to tell the compiler how you want the program to run by focussing on the logic implementation. Declarative code, on the other hand enables you to describe the logic without worrying about the implementation. How does it help? You do not need to invest time in dealing with how your program runs and deciding what to load first.
+
+Lets see a working example. 
+
+With jQuery you might write the following code to define a textbox that takes numeric values and doesn’t allow alphanumeric input or input that is beyond of a defined range.
+
+```
+$("#textbox").numericTextBox({
+    value: 10,
+    min: -10,
+    max: 100,
+    step: 0.75,
+    format: "n",
+    decimals: 3
+});
+```
+
+<box type="info">
+If you are not familiar with jQuery, you may want to get a brief understanding from the official <a href="https://jquery.com/">website</a>.
+</box>
+
+On the other hand, using versions Angular 2 and above, your code would look something like this:
+
+```
+<input numerictextbox k-min="-10" k-max="100" k-step="0.75"
+  k-format="n" k-decimals="3"/>
+```
+
+Rather than spending time thinking on how the program should flow, you define what you want the user to see and Angular will take care of the dependencies.
+
+The above example shows how Angular makes it easier for developers to design the UI elements.
+
+[source](https://www.telerik.com/blogs/three-ds-of-web-development-1-declarative-vs-imperative)
+
```

Comment:
Overall, this benefit is not entirely clear. You start by saying `Angular uses HTML to define the app’s user interface.`. And the reader could wonder, 'as opposed to what? doesn't everybody use HTML to define the user interface?'